### PR TITLE
Function to serialize compressed image to buffer

### PR DESCRIPTION
**Pull request by mjcarroll**
Signed-off-by: Michael Carroll <michael@openrobotics.org>

### CODE REVIEW

**Review by chapulina — COMMENTED** [review]
Mind adding a quick test to check the buffer's contents?

File: graphics/src/Image.cc [hunk 1]
```diff
@@ -137,6 +137,19 @@ void Image::SavePNG(const std::string &_filename)
   FreeImage_Save(FIF_PNG, this->dataPtr->bitmap, _filename.c_str(), 0);
 }
 
+//////////////////////////////////////////////////
+void Image::SavePNGToBuffer(std::vector<unsigned char>& buffer)
+{
+  FIMEMORY *hmem = FreeImage_OpenMemory();
+  FreeImage_SaveToMemory(FIF_PNG, this->dataPtr->bitmap, hmem);
+  unsigned char* mem_buffer = nullptr;
+  unsigned int size_in_bytes = 0;
```
Comment (iche033) [inline graphics/src/Image.cc:146]: use camelCase?

**Review by iche033 — APPROVED** [review]
just minor comment on style otherwise looks good to me

File: graphics/src/Image.cc (hunk 1 above)
Comment (nkoenig) [reply graphics/src/Image.cc]: Fixed in bc1a0a6256ff194baf94630dd18929e457a49a6f

**Review by nkoenig — APPROVED** [review]

**Comment by iche033** [comment]
windows build caught a compile error:

```
C:\Jenkins\workspace\ignition_common-ci-pr_any-windows7-amd64\ws\ign-common\graphics\src\Image.cc(155): error C2664: 'BOOL FreeImage_AcquireMemory(FIMEMORY *,BYTE **,DWORD *)': cannot convert argument 3 from 'unsigned int *' to 'DWORD *'
```

**Comment by nkoenig** [comment]
@mjcarroll  and @iche033 , CI is now green.

**Review by iche033 — APPROVED** [review]

**Comment by mjcarroll** [comment]
@chapulina there isn't an easy way to test this without implementing the inverse (png buffer to image).  Since this is needed by subt, can we land it and I'll add that test in a followup?

**Comment by chapulina** [comment]
Sure sounds good. I was thinking of a simpler test, maybe just checking the size of the buffer. To make sure that the function is exercised in the tests. But don't block on this.

**Pull request merged**